pkg/apis/autoscaling/v1alpha1/cronfederatedhpa_types.go

```diff
@@ -0,0 +1,205 @@
+	TargetMinReplicas *int32 `json:"targetMinReplicas,omitempty"`
+
+	// TargetMaxReplicas is the target MaxReplicas to be set for FederatedHPA.
+	// nil means the MaxReplicas will not be updated and in this case
+	// TargetMinReplicas must be specified if referencing resource is FederatedHPA.
+	// +optional
+	TargetMaxReplicas *int32 `json:"targetMaxReplicas,omitempty"`
+
+	// Suspend tells the controller to suspend subsequent executions.
+	// Defaults to false.
+	//
+	// +kubebuilder:default=false
+	// +optional
+	Suspend *bool `json:"suspend,omitempty"`
+
+	// TimeZone for the giving schedule.
+	// If not specified, this will default to the time zone of the
+	// karmada-controller-manager process.
+	// Invalid TimeZone will be rejected when applying by karmada-webhook.
+	// see https://en.wikipedia.org/wiki/List_of_tz_database_time_zones for the
+	// all timezones.
+	// +optional
+	TimeZone *string `json:"timeZone,omitempty"`
+
+	// SuccessfulHistoryLimit represents the count of successful execution items
+	// for each rule.
+	// The value must be a positive integer. It defaults to 3.
+	//
+	// +kubebuilder:default=3
+	// +kubebuilder:validation:ExclusiveMinimum=0
+	// +kubebuilder:validation:ExclusiveMaximum=32
+	// +optional
+	SuccessfulHistoryLimit *int32 `json:"successfulHistoryLimit,omitempty"`
+
+	// FailedHistoryLimit represents the count of failed execution items for
+	// each rule.
+	// The value must be a positive integer. It defaults to 3.
+	//
+	// +kubebuilder:default=3
+	// +kubebuilder:validation:ExclusiveMinimum=0
+	// +kubebuilder:validation:ExclusiveMaximum=32
+	FailedHistoryLimit *int32 `json:"failedHistoryLimit,omitempty"`
+}
+
+// CronFederatedHPAStatus represents the current status of a CronFederatedHPA.
+type CronFederatedHPAStatus struct {
+	// ExecutionHistories record the execution histories of CronFederatedHPARule.
+	// +optional
+	ExecutionHistories []ExecutionHistory `json:"executionHistories,omitempty"`
+}
+
+// ExecutionHistory records the execution history of specific CronFederatedHPARule.
+type ExecutionHistory struct {
+	// RuleName is the name of the CronFederatedHPARule.
+	// +required
+	RuleName string `json:"ruleName"`
+
+	// NextExecutionTime is the next time to execute.
+	// Nil means the rule has been suspended.
+	// +optional
+	NextExecutionTime *metav1.Time `json:"nextExecutionTime,omitempty"`
+
+	// SuccessfulExecutions records successful executions.
+	// +optional
+	SuccessfulExecutions []SuccessfulExecution `json:"successfulExecutions,omitempty"`
+
+	// FailedExecutions records failed executions.
+	// +optional
+	FailedExecutions []FailedExecution `json:"failedExecutions,omitempty"`
+}
+
+// SuccessfulExecution records a successful execution.
+type SuccessfulExecution struct {
+	// ScheduleTime records the scheduled time.
+	// +required
+	ScheduleTime *metav1.Time `json:"scheduleTime"`
+
+	// ExecutionTime records the execution time.
+	// +required
+	ExecutionTime *metav1.Time `json:"executionTime"`
```

Comment:
What is the difference between `ExecutionTime` and `ScheduleTime`?

Reply:
Updated in the comments.

```
ScheduleTime is the `expected` execution time declared in CronFederatedHPARule.

ExecutionTime is the `actual` execution time of CronFederatedHPARule.  Tasks may not always be executed at ScheduleTime. ExecutionTime is used to evaluate the efficiency of the controller's execution.
```